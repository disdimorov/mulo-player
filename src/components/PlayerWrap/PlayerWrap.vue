<template>
  <div class="player-wrap">
    <img alt="App logo" src="@/assets/logo.svg" width="256" height="256">
    <Player :file="URL"/>
  </div>
</template>

<script>
import Player from '@/views/Player/Player.vue';

export default {
  name: 'Auth',
  components: { Player },

  data() {
    return { URL: 'https://dl4.ru-music.xn--41a.ws/mp3/1584.mp3' };
  },
};
</script>

<style scoped lang="less">
.player-wrap {}
</style>
