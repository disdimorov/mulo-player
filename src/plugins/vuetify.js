import Vue from 'vue';
import Vuetify from 'vuetify';
import { THEME as theme } from '@/const';

Vue.use(Vuetify);

export default new Vuetify({ theme });
