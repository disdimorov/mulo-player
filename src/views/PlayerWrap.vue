<template>
  <div class="player-wrap">
    <img alt="App logo" src="../assets/logo.svg">
    <Player/>
  </div>
</template>

<script>
import Player from '../components/Player.vue';

// TODO replace logo
// TODO add unit
export default {
  name: 'Auth',
  components: { Player },
};
</script>

<style scoped lang="less">
.player-wrap {}
</style>
