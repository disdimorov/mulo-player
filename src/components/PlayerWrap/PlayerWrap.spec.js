import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import vuetify from '@/plugins/vuetify';
import PlayerWrap from './PlayerWrap.vue';

const localVue = createLocalVue();

describe('Component PlayerWrap', () => {
  const createWrapper = ({ isWithoutStubs = false, ...options } = {}) => {
    const renderer = isWithoutStubs ? mount : shallowMount;
    const wrapper = renderer(PlayerWrap, {
      localVue,
      vuetify,
      ...options,
    });

    return { wrapper };
  };

  describe('snapshots', () => {
    it('should match snapshot', () => {
      const { wrapper } = createWrapper({ isWithoutStubs: true });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
