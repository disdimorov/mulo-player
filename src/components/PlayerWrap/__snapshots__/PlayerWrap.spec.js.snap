// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PlayerWrap snapshots should match snapshot 1`] = `
<div class="player-wrap"><img alt="App logo" src="@/assets/logo.svg" width="256" height="256">
  <div class="player"><button title="Stop" type="button" class="control"><img src="" alt="stop icon" width="20" height="20">
      <!----> </button> <button title="Play/Pause" type="button" class="control"><img src="" alt="pause icon" width="20" height="20"> <img src="" alt="play icon" width="20" height="20" style="display: none;"> </button>
    <div class="time-line">
      <div title="Time played : Total time" class="time-line__progress">
        <div class="time-line__line"></div>
      </div>
      <div class="time-line__timers">
        <div class="time-line__current">00:00</div>
        <div class="time-line__total">00:00</div>
      </div>
    </div> <button title="Download" type="button" class="control"><img src="" alt="download icon" width="20" height="20">
      <!----> </button> <button title="Loop/Once" type="button" class="control"><img src="" alt="once icon" width="20" height="20"> <img src="" alt="loop icon" width="20" height="20" style="display: none;"> </button> <button title="Mute" type="button" class="control"><img src="" alt="sound icon" width="20" height="20"> <img src="" alt="mute icon" width="20" height="20" style="display: none;"> </button> <button title="Volume" type="button" class="control"><img src="" alt="volume icon" width="20" height="20">
      <!----> <input type="range" min="0" step="0.01" max="1" style="display: none;"></button>
  </div>
</div>
`;
